src/index_factory.cpp

```diff
@@ -0,0 +1,121 @@
+#include "index_factory.h"
+
+namespace diskann
+{
+
+IndexFactory::IndexFactory(IndexConfig &config) : _config(config)
+{
+    check_config();
+}
+
+std::unique_ptr<AbstractIndex> IndexFactory::instance()
```

Comment:
`Index` supports `uint64_t` for the tag type. Do we support it here? Do we need a config for tag type?